src/formatDate.js

```diff
@@ -42,15 +42,31 @@
  *   ['DD', 'MM', 'YYYY', '.'],
  * ) // '18.02.1997'
  *
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
+  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const dateCollector = {};
+  const newDateFormat = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateCollector[fromFormat[i]] = dateParts[i];
+  }
+
+  if (!fromFormat.includes("YY")) {
+    dateCollector.YY = dateCollector.YYYY.slice(2);
+  } else {
+    const longYear =
+      dateCollector.YY < 30 ? 20 + dateCollector.YY : 19 + dateCollector.YY;
```

Comment:
Consider concatenating strings using template literals instead of the '+' operator. For example, `20${dateCollector.YY}` instead of `20 + dateCollector.YY`.